test/api.test.js

```diff
@@ -535,6 +546,68 @@ describe("api.js", () => {
             }]);
         });
 
+        it("should fail by timeout during rendition processing", async () => {
+            delete process.env.DISABLE_IO_EVENTS_ON_TIMEOUT;
+            process.env.ASSET_COMPUTE_SDK_DISABLE_CGROUP_METRICS = true;
+
+            const receivedMetrics = MetricsTestHelper.mockNewRelic();
+            testUtil.nockIOEvent({
+                type: "rendition_failed",
+                errorReason: "GenericError",
+                rendition: {
+                    fmt: "png"
+                },
+                source: "https://example.com/MySourceFile.jpg"
+            });
+
+            const main = worker(async function() {
+                await sleep(200);
+            });
+            assert.equal(typeof main, "function");
+            process.env.__OW_DEADLINE = Date.now() + 100;
+
+            await main(testUtil.simpleParams({noEventsNock:true, noPut:true}));
+            testUtil.assertNockDone();
+            await MetricsTestHelper.metricsDone();
+            MetricsTestHelper.assertArrayContains(receivedMetrics, [{
+                eventType: 'error',
+                location: 'test_action_timeout'
+            }]);
```

Comment:
As mentioned above, we still want the `timeout` metric, so should assert it in the tests here.